tests/sentry/ingest/billing_metrics_consumer/test_billing_metrics_consumer_kafka.py

```diff
@@ -91,33 +96,49 @@ def generate_kafka_message(bucket: MetricsBucket) -> Message[KafkaPayload]:
     # then call submit when there is a message.
     strategy.poll()
     strategy.poll()
-    assert track_outcome.call_count == 0
-    for i, bucket in enumerate(buckets):
+
+    for bucket in buckets:
         strategy.poll()
         strategy.submit(generate_kafka_message(bucket))
-        # commit is called for every two messages:
-        assert fake_commit.call_count == i // 2
-        if i < 3:
-            assert track_outcome.call_count == 0
-        else:
-            assert track_outcome.mock_calls == [
-                mock.call(
-                    org_id=1,
-                    project_id=2,
-                    key_id=None,
-                    outcome=Outcome.ACCEPTED,
-                    reason=None,
-                    timestamp=datetime(1970, 1, 2, 10, 17, 36),
-                    event_id=None,
-                    category=DataCategory.TRANSACTION_PROCESSED,
-                    quantity=3,
-                )
-            ]
 
-    # There's been 5 messages, 2 x 2 of them have their offsets committed:
-    assert fake_commit.call_count == 2
+    assert strategy._billing_producer.produce.call_count == 1
+    called_payload = KafkaPayload(
+        key=None,
+        value=json.dumps(
+            {
+                "timestamp": time,
+                "org_id": 1,
+                "project_id": 2,
+                "outcome": 0,
+                "category": 8,
+                "quantity": 3,
+            }
+        ).encode("utf-8"),
+        headers=[],
+    )
+    assert strategy._billing_producer.produce.mock_calls[0] == mock.call(
+        destination=billing_topic,
+        payload=called_payload,
+    )
```

Comment:
I have not asserted on the whole `mock_calls` because there are a bunch of calls from magic methods that would make the test easier to fail (e.g. on arroyo's internal changes); and add noise, making the test more difficult to read and focus on the important parts. There's an assertion above to ensure there's only one call we've made, so we shouldn't expect more items in the list to be relevant.